app/src/main/kotlin/compiler/intermediate/Constant.kt

```diff
@@ -15,3 +15,11 @@ class SummedConstant(private val first: Constant, private val second: Constant)
 
     override val value get() = first.value + second.value
 }
+
+class ConstantAlignedToAGivenRestModulo(private val rawValue: Constant, private val modulo: Long, private val rest: Long = 0L) : Constant {
```

Comment:
I think the word "remainder" is used instead of "rest" in this context.

Reply:
I'd suggest to simply name this as `AlignedConstant`.

Reply:
You're right. Applied both suggestions and added a short doc comment